src/test/java/com/drajer/test/ITBaseCustom.java

```diff
@@ -0,0 +1,117 @@
+  String uri = "/api/launchPatient";
+
+  private String testCaseId;
+  private Map<String, String> testData;
+  private Map<String, ?> allResourceMapping;
+  private Map<String, ?> allOtherMapping;
+
+  public ITBaseCustom(
+      String testCaseId,
+      Map<String, String> testData,
+      Map<String, ?> resourceMapping,
+      Map<String, ?> otherMapping) {
+    this.testCaseId = testCaseId;
+    this.testData = testData;
+    this.allResourceMapping = resourceMapping;
+    this.allOtherMapping = otherMapping;
+  }
+
+  private static final Logger logger = LoggerFactory.getLogger(ITBaseCustom.class);
+  private String systemLaunch3PayLoad;
+  WireMockHelper stubHelper;
+
+  @Before
+  public void launchTestSetUp() throws IOException {
+    logger.info("Executing test: {}", testCaseId);
+    tx = session.beginTransaction();
+
+    // Data Setup
+    saveHealtcareSetting(testData.get("HealcareSettingsFile"));
+
+    systemLaunch3PayLoad = getSystemLaunch3Payload(testData.get("SystemLaunch3Payload"));
+    session.flush();
+    tx.commit();
+
+    stubHelper = new WireMockHelper(wireMockServer, wireMockHttpPort);
+    logger.info("Creating WireMock stubs..");
+    stubHelper.stubResources(allResourceMapping);
+    stubHelper.stubAuthAndMetadata(allOtherMapping);
+    mockRestApi();
+  }
+
+  @Parameters(name = "{0}")
+  public static Collection<Object[]> data() {
+    TestDataGenerator testDataGenerator = new TestDataGenerator("test-yaml/customQueryTest.yaml");
+
+    TestDataGenerator gen;
+
+    Set<String> testCaseSet = testDataGenerator.getAllTestCases();
+    Object[][] data = new Object[testCaseSet.size()][4];
+
+    int count = 0;
+    for (String testCase : testCaseSet) {
+      data[count][0] = testCase;
+      data[count][1] = testDataGenerator.getTestCaseByID(testCase).getTestData();
+      data[count][2] = testDataGenerator.getResourceMappings(testCase);
+      data[count][3] = testDataGenerator.getOtherMappings(testCase);
+      count++;
+    }
+    return Arrays.asList(data);
+  }
+
+  @Test
+  public void callApi() {
+    wireMockServer.verify(0, getRequestedFor(urlEqualTo("/FHIR/Encounter?patient=12742571")));
+    wireMockServer.verify(0, getRequestedFor(urlEqualTo("/FHIR/Patient?patient=12742571")));
+    final TestRestTemplate restTemplate = new TestRestTemplate();
+    HttpHeaders headers = new HttpHeaders();
+    String requestId = "1234";
+    String url = createURLWithPort(uri);
+    headers.setContentType(MediaType.APPLICATION_JSON);
+    headers.add("X-Request-ID", requestId);
+    HttpEntity<String> entity = new HttpEntity<>(systemLaunch3PayLoad, headers);
+    ResponseEntity<String> response =
+        restTemplate.exchange(url, HttpMethod.POST, entity, String.class);
+    response.getStatusCode();
+    wireMockServer.verify(1, getRequestedFor(urlEqualTo("/FHIR/Encounter?patient=12742571")));
+    wireMockServer.verify(1, getRequestedFor(urlEqualTo("/FHIR/Patient?patient=12742571")));
```

Comment:
These call can occur more then one time in the workflow, change it as below.
```suggestion
    wireMockServer.verify(moreThanOrExactly(1), getRequestedFor(urlEqualTo("/FHIR/Encounter?patient=12742571")));
    wireMockServer.verify(moreThanOrExactly(1), getRequestedFor(urlEqualTo("/FHIR/Patient?patient=12742571")));
```